File: sales-portal-robot-tests/exmple_js_project/src/data/schemas/users/user.schema.ts
import { obligatoryFieldsSchema, obligatoryRequiredFields } from "../core.schema";
import { ROLES } from "data/types/user.types";

export const userSchema = {
  type: "object",
  properties: {
    _id: { type: "string" },
    username: { type: "string" },
    firstName: { type: "string" },
    lastName: { type: "string" },
    roles: {
      type: "array",
      items: {
        type: "string",
        enum: Object.values(ROLES),
      },
    },
    createdOn: { type: "string" },
  },
  required: ["_id", "username", "firstName", "lastName", "roles", "createdOn"],
  additionalProperties: false,
};

export const getUserSchema = {
  type: "object",
  properties: {
    User: userSchema,
    ...obligatoryFieldsSchema,
  },
  required: ["User", ...obligatoryRequiredFields],
  additionalProperties: false,
};

export const getAllUsersSchema = {
  type: "object",
  properties: {
    Users: {
      type: "array",
      items: userSchema,
    },
    ...obligatoryFieldsSchema,
  },
  required: ["Users", ...obligatoryRequiredFields],
  additionalProperties: false,
};

File: sales-portal-robot-tests/data/schemas/users/user_schema.py
"""User JSON schemas."""

from __future__ import annotations

from typing import Any

from data.schemas.core_schema import OBLIGATORY_FIELDS_SCHEMA, OBLIGATORY_REQUIRED_FIELDS

USER_SCHEMA: dict[str, Any] = {
    "type": "object",
    "properties": {
        "_id": {"type": "string"},
        "username": {"type": "string"},
        "firstName": {"type": "string"},
        "lastName": {"type": "string"},
        "roles": {
            "type": "array",
            "items": {
                "type": "string",
                "enum": ["ADMIN", "USER"],
            },
        },
        "createdOn": {"type": "string"},
    },
    "required": ["_id", "username", "firstName", "lastName", "roles", "createdOn"],
    "additionalProperties": False,
}

GET_USER_SCHEMA: dict[str, Any] = {
    "type": "object",
    "properties": {
        "User": USER_SCHEMA,
        **OBLIGATORY_FIELDS_SCHEMA,
    },
    "required": ["User", *OBLIGATORY_REQUIRED_FIELDS],
    "additionalProperties": False,
}

GET_ALL_USERS_SCHEMA: dict[str, Any] = {
    "type": "object",
    "properties": {
        "Users": {
            "type": "array",
            "items": USER_SCHEMA,
        },
        **OBLIGATORY_FIELDS_SCHEMA,
    },
    "required": ["Users", *OBLIGATORY_REQUIRED_FIELDS],
    "additionalProperties": False,
}
